sdks/python/apache_beam/runners/worker/operations.py

```diff
@@ -299,7 +299,14 @@ def _read_side_inputs(self, tags_and_types):
         if not isinstance(si, operation_specs.WorkerSideInputSource):
           raise NotImplementedError('Unknown side input type: %r' % si)
         sources.append(si.source)
-      iterator_fn = sideinputs.get_iterator_fn_for_sources(sources)
+      si_counter = opcounters.SideInputReadCounter(
```

Comment:
<!--thread_id:cc_158386111_t; commit:0bfc12f8986238da5b5014cfc5eea4d32e6638f7; resolved:1-->
<!--section:context-quote-->
> **charlesccychen** wrote:
> It doesn't look like to me that the changes here are guarded by the experiment flag?  Perhaps we can run a quick profile to verify these new changes don't change performance (particularly with the "with self.read_counter" in side_inputs.py).

<!--section:body-->
You're right. I've added them now - and I'll run the profile tests in a bit. Marking as Done to track in comment within sideinputs.py.